File: src/2025-03-04-03-28-40--deepseek-chat-0304/go/exercises/practice/transpose/transpose.go
package transpose

func Transpose(input []string) []string {
	if len(input) == 0 {
		return []string{}
	}

	// Find maximum row length
	maxLen := 0
	for _, row := range input {
		if len(row) > maxLen {
			maxLen = len(row)
		}
	}

	// Build transposed matrix
	result := make([]string, maxLen)
	for i := 0; i < maxLen; i++ {
		var transposedRow []byte
		for rowIdx := 0; rowIdx < len(input); rowIdx++ {
			if i < len(input[rowIdx]) {
				transposedRow = append(transposedRow, input[rowIdx][i])
			} else if len(transposedRow) > 0 {
				// Only pad with space if we've already started building this row
				transposedRow = append(transposedRow, ' ')
			}
		}
		result[i] = string(transposedRow)
	}

	return result
}

File: src/2025-03-04-03-28-40--deepseek-chat-0304/python/exercises/practice/transpose/transpose.py
def transpose(text):
    """Transpose text, handling rows of different lengths by left-padding with spaces."""
    if not text:
        return ""
    
    # Split into lines and find max length
    lines = text.split('\n')
    max_len = max(len(line) for line in lines)
    
    # Pad lines with spaces on the right to make them equal length
    padded_lines = [line.ljust(max_len) for line in lines]
    
    # Transpose the padded lines
    transposed = []
    for i in range(max_len):
        # Build each column as a new row
        new_row = ''.join(line[i] for line in padded_lines)
        transposed.append(new_row)
    
    # Join with newlines
    result = '\n'.join(transposed)
    return result
